sap_hana: add Data Observability RC query actions support (#23965)
DataDog/integrations-core: sap_hana/README.md

See the [sample sap_hana.d/conf.yaml][4] for all available schema collection options, including `include_schemas` and `exclude_schemas`.

#### Data Observability query actions

The Agent can execute Data Quality Monitor queries on your SAP HANA instance on behalf of Datadog. Queries are delivered via Remote Configuration - no manual query configuration is required. This allows Data Quality Monitors to run directly on agent-monitored instances that are not reachable from the Datadog cloud.

> To enable query actions, add the following to your `sap_hana.d/conf.yaml` instance block:

Author: The remote_configuration.agent_integrations.allow_list defaults to an empty list in the Datadog Agent, which means RC-delivered integration configs are silently rejected by default. To allow the RC backend to push
  data_observability query configs to the sap_hana check, you must explicitly add sap_hana to the allow list in datadog.yaml:
```
  remote_configuration:
    agent_integrations:
      allow_list:
        - sap_hana
```
  Without this, any DO query dispatched by the backend to this agent is dropped with "Integration sap_hana is not allowed to be scheduled in this agent" — no error is surfaced to the user, the agent just silently skips it.

See the [sample sap_hana.d/conf.yaml][4] for all available schema collection options, including `include_schemas` and `exclude_schemas`.

#### Data Observability query actions

The Datadog backend can deliver monitoring queries to the SAP HANA check via Remote Configuration. When enabled, the Agent executes these queries against HANA on a schedule and forwards the results as Data Observability events.

> To allow Remote Configuration to push query configs to the `sap_hana` check, add it to the allow list in `datadog.yaml`:

Reviewer: ```suggestion
To allow Remote Configuration to push query configs to the `sap_hana` check, add `sap_hana` to the allowlist in `datadog.yaml`:
```

```yaml
remote_configuration:
  agent_integrations:
    allow_list:
      - sap_hana
```

> Without this entry, the Agent silently drops any query dispatched by the backend with no error surfaced. After updating `datadog.yaml`, [restart the Agent][5].

Reviewer: ```suggestion
Without this entry, the Agent silently drops any query delivered by the backend without surfacing an error. After updating `datadog.yaml`, [restart the Agent][5].
```

> Data Observability query actions require schema collection to be enabled. Ensure the `collect_schemas` block is present and `enabled: true` in your `sap_hana.d/conf.yaml` (see [Schema collection](#schema-collection)).

Reviewer: ```suggestion
Data Observability query actions require schema collection to be enabled. Verify that the `collect_schemas` block is present and `enabled: true` in your `sap_hana.d/conf.yaml` (see [Schema collection](#schema-collection)).
```

See the [sample sap_hana.d/conf.yaml][4] for all available schema collection options, including `include_schemas` and `exclude_schemas`.

#### Data Observability query actions

> The Datadog backend can deliver monitoring queries to the SAP HANA check via Remote Configuration. When enabled, the Agent executes these queries against HANA on a schedule and forwards the results as Data Observability events.

Reviewer: ```suggestion
The Datadog backend can deliver monitoring queries to the SAP HANA check through Remote Configuration. When enabled, the Agent executes these queries against HANA on a schedule and forwards the results as Data Observability events.
```
